### PR TITLE
add basic cli program wrapping the library

### PR DESCRIPTION
**Pull request by erykpiast**

### CODE REVIEW

**Comment by bcherny** [comment]
Thanks for the contribution @erykpiast! How is this meant to be used? `cat foo.json | json2ts > out.ts`? Please add some docs if this is the case.

**Comment by erykpiast** [comment]
It's not only stdin/stdout, but parameters and named parameters too. I've updated `README`.

File: README.md [hunk 1]
```diff
@@ -63,6 +63,19 @@ fs.writeFileSync('dist/foo.d.ts', await compileFromFile('src/foo.json'))
 
 See [/example](example) for a fully working demo.
 
+### CLI
+Simple CLI utility is provided within the package.
+```
+cat foo.json | json2ts > foo.d.ts
+# or
+json2ts foo.json foo.d.ts
```
Comment (bcherny) [inline README.md]: Can we default `json2ts foo.json` to output foo.d.ts in the same directory as foo.json?

File: src/cli.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+})
+const argIn: string = argv._[0] || argv['input']
+const argOut: string = argv._[1] || argv['output']
+
+function readInput(): Promise<string> {
+  if (!argIn) {
+    return new Promise((resolve, reject) => {
+      stdin((str: string) => {
+        try {
+          resolve(str)
```
Comment (bcherny) [inline src/cli.ts]: When would this throw?

File: src/cli.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    })
+  }
+
+  return readFile(argIn)
+}
+
+function writeOutput(compiled: string): Promise<void> {
+  if (!argOut) {
+    try {
+      process.stdout.write(compiled)
```
Comment (bcherny) [inline src/cli.ts]: When would this throw?

File: src/cli.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+const argv: minimist.ParsedArgs = minimist(process.argv.slice(2), {
+  alias: {
+    input: [ 'i' ],
+    output: [ 'o' ]
+  }
+})
+const argIn: string = argv._[0] || argv['input']
+const argOut: string = argv._[1] || argv['output']
+
+function readInput(): Promise<string> {
```
Comment (bcherny) [inline src/cli.ts]: Can make this `async`

File: README.md (hunk 1 above)
Comment (erykpiast) [inline README.md]: :+1:

File: src/cli.ts (hunk 3 above)
Comment (erykpiast) [inline src/cli.ts]: I think it is possible to get error when writing to stdout, isn't it?

File: src/cli.ts (hunk 2 above)
Comment (erykpiast) [inline src/cli.ts]: never, it doesn't make sense :D

File: src/cli.ts (hunk 4 above)
Comment (erykpiast) [inline src/cli.ts]: I'm not sure it does make sense. The code could be simplified to
```
function readInput(): Promise<string> {
  if (!argIn) {
    return new Promise(stdin);
  }

  return readFile(argIn)
}
```
Is it the case for `async`?

**Comment by bcherny** [comment]
Looking good- thanks @erykpiast!

**Pull request merged**